src/coreclr/gc/windows/gcenv.windows.cpp

```diff
@@ -434,6 +445,37 @@ size_t GetLogicalProcessorCacheSizeFromOS()
     if(pslpi)
         delete[] pslpi;  // release the memory allocated for the SLPI array.
 
+#if defined(TARGET_ARM64)
+    if (cache_level != 3)
+    {
+        // We expect to get the L3 cache size for Arm64 but currently expected to be missing that info
+        // from most of the machines.
+        // Hence, just use the following heuristics at best depending on the CPU count
+        // 1 ~ 4   :  4 MB
+        // 5 ~ 16  :  8 MB
+        // 17 ~ 64 : 16 MB
+        // 65+     : 32 MB
+        if (totalCPUCount < 5)
```

Comment:
Here and elsewhere, should this be physical cores or logical cores and should it take into account affinity?

I tested this on Ampere Arm64 windows machine where we cannot fetch L3 cache size. There are 2 CPU groups with 16 / 64 cores and even if I set `COMPlus_GCCpuGroup=1` and `COMPlus_Thread_UseAllCpuGroups=1` to use all cores, the `totalCPUCount` would switch between 16 and 64 and depending on that would give 8 MB or 16 MB.

Reply:
you should be able to use `GCToEEInterface::GetCurrentProcessCpuCount` to get the total CPU count this process can use

Reply:
I verified the results with `GCToOSInterface::GetTotalProcessorCount` since I can't use `GCToEEInterface::GetCurrentProcessCpuCount` at that site.

I see some discrepancies in using `GCToEEInterface::GetCurrentProcessCpuCount` vs. what I have currently. If I just print `pslcount` (the one I am using based on `RelationProcessorCore` vs. the one given by `GCToOSInterface::GetTotalProcessorCount` vs. `Environment.ProcessorCount`, I see that `pslcount` and `Environment.ProcessorCount` matches and should be used for consistency. Thoughts?

```
Environment.ProcessorCount = 64
pslcount= 64, totalCpuCount= 16, cache_size= 8 MB
```

Reply:
Spoke offline and we will go ahead with `GetCurrentProcessCpuCount()` because that returns correct values when `
COMPlus_GCCpuGroup=1` and `COMPlus_Thread_UseAllCpuGroups=1` are set.